node/builder_miner.go

```diff
@@ -141,7 +143,7 @@ func ConfigStorageMiner(c interface{}) Option {
 		Override(new(config.HarmonyDB), cfg.HarmonyDB),
 		Override(new(harmonydb.ITestID), harmonydb.ITestID("")),
 		Override(new(*ctladdr.AddressSelector), modules.AddressSelector(&cfg.Addresses)),
-		If(build.F3Enabled, Override(F3Participation, modules.F3Participation)),
+		If(build.F3Enabled && os.Getenv(lf3.F3DisableEnvKey) != lf3.F3DisableEnvValue, Override(F3Participation, modules.F3Participation)),
```

Comment:
@adlrocha I think we need a similar check to disable subscribing to the F3 Gossip topic if this env var is disabled.
I think a  goroutine that does this periodically makes sense there too.

Reply:
There is no reason for additional goroutine or anything. If the envvar is set, then F3 won't get constructed by the dependency injection. I'm assuming that we don't change this envvar dynamically during runtime.

Reply:
I don't know if pubsub would relay messages for topics that are allowed by the subscription filter, _but not subscribed to_. 

In either case I think it is better to not even allow the topic if F3 is disabled. That way nothing can possibly go wrong.

Reply:
Yeah, although that is separate from trying to close F3 after envvar is changed. We should probably create `IsF3Enabled()` in `build` use it in the lp2p/pubsub.go and dep injection.

Reply:
Extended the env var check to pubsub too, I completely missed this one. That being said, there's not much we can do gossipsub-wise without restarting the node if we disable F3 in runtime, as pubsub filtering is a constructor option (AFAIK). Please correct me if I am missing something.

Reply:
Message relaying will not happen until join/subscribe is called on a topic. Both of those happen deep inside go-f3 once the F3 is instantiated, which means having the topic in allowed list or not does not matter. 

I think this is not a blocker. I defer to @Kubuxu and @aarshkshah1992.